lib/instrumentation/grpc-js/grpc.js

```diff
@@ -6,80 +6,219 @@
-          })
-        } else {
-          shim.logger.debug('Distributed tracing disabled by instrumentation.')
+        segment.addAttribute('grpc.statusCode', code)
+        segment.addAttribute('grpc.statusText', details)
+
+        // Java captures client errors based on status code
+        // but has specific gRPC configuration to turn off
+        if (code !== 0) {
+          // this is currently just creating an error from the details string
+          shim.agent.errors.add(segment.transaction, details)
         }
 
-        args[0] = nrMetadata
+        segment.addAttribute('component', 'gRPC')
 
-        const originalListener = args[1]
-        const nrListener = Object.assign({}, originalListener)
-        nrListener.onReceiveStatus = (status) => {
-          const { code, details } = status
+        const protocol = 'grpc'
 
-          segment.addAttribute('grpc.statusCode', code)
-          segment.addAttribute('grpc.statusText', details)
+        const url = `${protocol}://${authorityName}${this.methodName}`
 
-          // Java captures client errors based on status code
-          // but has specific gRPC configuration to turn off
-          if (code !== 0) {
-            // this is currently just creating an error from the details string
-            shim.agent.errors.add(segment.transaction, details)
-          }
+        segment.addAttribute('http.url', url)
+        segment.addAttribute('http.method', this.methodName)
 
-          segment.addAttribute('component', 'gRPC')
+        segment.end()
 
-          const protocol = 'grpc'
+        originalListener && originalListener.onReceiveStatus(status)
+      }
 
-          const url = `${protocol}://${authorityName}${this.methodName}`
+      args[1] = nrListener
 
-          segment.addAttribute('http.url', url)
-          segment.addAttribute('http.method', this.methodName)
+      return original.apply(this, args)
+    }
+  }
+}
 
-          segment.end()
+/**
+ * Instruments the grpc-js server by intercepting the moment when
+ * server methods are registered from the method implementations
+ * provided to grpc-js. This handles all four types of server
+ * invocations: unary, client-streaming, server-streaming, and
+ * bidirectional streaming.
+ *
+ * @param {object} shim the web shim to instrument with
+ * @param {Function} original the original function
+ * @returns {Function} the instrumented function
+ */
+function wrapRegister(shim, original) {
+  const constants = shim.require('./build/src/constants')
+
+  return function wrappedRegister() {
+    const args = shim.argsToArray.apply(shim, arguments)
+
+    const name = args[0]
+    const handler = args[1]
+    const type = args[args.length - 1]
+    // We need the server for the host and port, but depending on how
+    // the user calls server functions, the connection might not be
+    // ready yet. It will be ready below by the time the registered
+    // handler is invoked.
+    const server = this
```

Comment:
this is no longer being used, oops and eslint and telling you that 😄

Reply:
Ugh, how did I miss that